source/04_best_practice/env_wrapper.rst

```diff
@@ -0,0 +1,118 @@
+
+Therefore, if the original environment is not perfectly adapted to our needs, we need to add functional modules to it to extend the functionality of the original environment and make it easy for the user to manipulate or adapt the environment's inputs and outputs. Env Wrapper is a simple solution for adding functionality.
+
+
+Env Wrapper offered by DI-engine
+==============================================
+
+DI-engine provides a large number of defined and generic Env Wrapper，The user can wrap it directly on top of the environment they need to use according to their needs.In the process of implementation，we refered  `OpenAI Baselines <https://github.com/openai/baselines/blob/master/baselines/common/atari_wrappers.py>`_ ，and folloiw the form of gym.Wrapper，which is `Gym.Wrapper <https://www.gymlibrary.dev/api/wrappers/>`_ ，In total, these include the following:
+
+- NoopResetEnv：add a reset method to the environment. Resets the environment after some no-operations..
+
+- MaxAndSkipEnv： Each ``skip`` frame（doing the same action）returns the maximum value of the two most recent frames.(for max pooling across time steps)。
+
+- WarpFrame： Convert the size of the image frame to 84x84, such as `Nature‘s paper <https://www.deepmind.com/publications/human-level-control-through-deep-reinforcement-learning>`_  and it's following work。(Note that this registrar also converts RGB images to GREY images)
+
+- ScaledFloatFrame： Normalize status values to 0~1。
+
+- ClipRewardEnv： Cuts the reward to {+1, 0, -1} by the positive or negative of the reward.
+
+- FrameStack： Set the nearest state frame of the stacked n_frames to the current state.
+
+- ObsTransposeWrapper：The dimensions of the observed state are adjusted to place the channel dimension on the first dimension of the state. It is typically used in atari environments.
+
+- RunningMeanStd：a wrapper for updating variances, means and counts.
+
+- ObsNormEnv：The observed states are normalised according to the running mean and std.
+
+- RewardNormEnv： The environmental rewards are normalised according to the standard deviation of the runs (running mean and std).
+
+- RamWrapper： Converting the ram state of the original environment into an image-like state by extending the dimensionality of the observed state.
+
+- EpisodicLifeEnv：Let the death of an intelligence in the environment mark the end of an episode (game over), and only reset the game when the real game is over. In general, this helps the algorithm to estimate the value.
+
+- FireResetEnv：  Take ``fire`` action when the environment is reset. For more information <https://github.com/openai/baselines/issues/240>
```

Comment:
format problem :For more information <https://github.com/openai/baselines/issues/240>